### PR TITLE
add EitherNev alias and syntax

### PR DESCRIPTION
**Pull request by key-eugene**
I'm using Vector a lot in my 2.13.7 codebase since it's one of the best collections right now and we either need to have those aliases in the business code or convert to List/Chanin in order to use cats' syntax. 
After asking if it exists somewhere @armanbilge [said](https://discord.com/channels/632277896739946517/632278570512678923/1012020048371662890) we can add it to cats.

### CODE REVIEW

**Comment by armanbilge** [comment]
CI doesn't seem to be running 🤔

**Comment by key-eugene** [comment]
Looks like CI finished 🙌

File: core/src/main/scala/cats/data/package.scala [hunk 1]
```diff
@@ -29,6 +29,7 @@ package object data extends ScalaVersionSpecificPackage {
   type IorNes[B, +A] = Ior[NonEmptySet[B], A]
   type EitherNel[+E, +A] = Either[NonEmptyList[E], A]
   type EitherNec[+E, +A] = Either[NonEmptyChain[E], A]
+  type EitherNev[+E, +A] = Either[NonEmptyVector[E], A]
```
Comment (johnynek) [inline core/src/main/scala/cats/data/package.scala:32]: for consistency should we add `IorNev` and `ValidatedNev`?

Comment (key-eugene) [reply core/src/main/scala/cats/data/package.scala:34]: good idea! sure!

**Comment by key-eugene** [comment]
Added both add `IorNev` and `ValidatedNev`

**Comment by johnynek** [comment]
actually... I wonder if we are moving in the wrong direction here.

Like, what if instead we did:

```scala
def toValidatedA[F[_]: Applicative](e: Either[A, B]): Validated[F[A], B] =
  e match {
    case Right(b) => Validated.valid(b)
    case Left(a) => Validated.invalid(Alternative[F].pure(a))
  }
```

then add a low priority typeclass instance for `Applicative[Validated[F[A], *]]` given a `SemigroupK[F]`.

Then all these instances would be the same.

**Comment by key-eugene** [comment]
Sure! I can prepare another MR or add to the existing one. What do you prefer?

**Comment by armanbilge** [comment]
@key-eugene are you still interested to work on Oscar's proposal? Seems like an interesting idea, I guess a separate PR would make more sense.

**Comment by satorg** [comment]
Yeah, I like that idea too. Seems similar to the discussion we have around generalization of `group`-alike methods in other PRs – instead of hardcoding output types we could try to elaborate on making them more generic.

**Comment by key-eugene** [comment]
@armanbilge yep! Should I close this one or we can merge it and I will work from there?

**Comment by key-eugene** [comment]
made another PR with proposed changes #4336

**Pull request closed without merging**